Sources/Extensions/SwiftStdlib/StringExtensions.swift

```diff
@@ -161,9 +161,10 @@ public extension String {
 	///		"john@doe.com".isEmail -> true
 	///
 	public var isEmail: Bool {
-		// http://stackoverflow.com/questions/25471114/how-to-validate-an-e-mail-address-in-swift
-		return range(of: "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}",
-					 options: .regularExpression, range: nil, locale: nil) != nil
+		guard self == trimmingCharacters(in: .whitespacesAndNewlines) else { return false }
```

Comment:
Instead of doing this you should just add `^` and `$` to the beginning and end of the regex, respectively. Doing that to the original version of this function would have solved #514 to begin with.

Reply:
Adding `^` and `$` solves the problem, however the old regex fails in examples like
`email.@.email.com` 🤔

Reply:
Apparently that's invalid, according to the list in @LucianoPAlmeida 's comment.

Reply:
@omaralbeik @guykogus 
Looks like that list is based [RFC 2822](https://tools.ietf.org/html/rfc2822) that is obsolete by [RFC 5322](https://tools.ietf.org/html/rfc5322#section-3.4.1), but anyways I believe this is an invalid email because  it seems like [RFC 5322](https://tools.ietf.org/html/rfc5322) explicitly only allows dots inside the local part of the mail address, not at the beginning and end.